File: 백준/Silver/1149. RGB거리/RGB거리.java
import java.io.*;
import java.util.*;

public class Main {
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        int N = Integer.parseInt(st.nextToken());
        int[] cur = new int[3];
        int[] next = new int[3];
        
    	for (int i = 0; i < N; i++){
    		st = new StringTokenizer(br.readLine());
    		for (int j=0;j<3;j++) {
    			next[j] = Integer.parseInt(st.nextToken());
    			next[j] += Math.min(cur[(j+1)%3], cur[(j+2)%3]);
    		}
    		for (int j = 0; j < 3; j++) cur[j] = next[j];
    	}
    	int min=Integer.MAX_VALUE;
    	for(int i=0;i<3;i++) min = Math.min(min,cur[i]);
    	System.out.println(min);
    }
    
}

File: 백준/Silver/1149. RGB거리/RGB거리.cc
#include <stdio.h>	
int MIN(int x, int y){
	return x < y ? x : y;
}
int main(){
	int n, s[3] = { 0, 0, 0 }, t[3], min;
	scanf("%d", &n);
	for (int i = 0; i < n; i++){
		scanf("%d %d %d", &t[0], &t[1], &t[2]);
		t[0] += MIN(s[1], s[2]);
		t[1] += MIN(s[0], s[2]);
		t[2] += MIN(s[0], s[1]);
		for (int j = 0; j < 3; j++){
			s[j] = t[j];
		}
	}
	min = s[0];
	for (int i = 1; i < 3; i++){
		if (s[i] < min){
			min = s[i];
		}
	}
	printf("%d", min);
	return 0;
}